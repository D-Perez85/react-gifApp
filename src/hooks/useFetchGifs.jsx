import { useState, useEffect } from "react";
import GetGifs from "../helpers/getGifs";
export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    GetGifs(category).then((images) => {
      setState({
        data: images,
        loading: false,
      });
    });
  }, [category]);

  return state; //{data: [],loading: false}
};
