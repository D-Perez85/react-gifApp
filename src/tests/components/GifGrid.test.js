import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import "@testing-library/jest-dom";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");
/*1. Mostrar el componente GifGrid correctamente 
    -shallow
    -wrapper
    -wrapper.toMatchSnapshot(); 
*/

describe("Pruebas en  <GifGrid/>", () => {
  useFetchGifs.mockReturnValue({
    data: [],
    loading: true,
  });

  const category = "One Punch";
  let wrapper = shallow(<GifGrid category={category} />);
  test("Snapshot de GifGrid ", () => {
    expect(wrapper).toMatchSnapshot();
  });


  test("Debe de mostrar items cuando se cargan imagenes UseFetch", () => {
    const gifs = [
      {
        id: "ABC",
        url: "Cualquier URL",
        title: "Cualquier Title",
      },
      {
        id: "ABd",
        url: "Cualquier URL",
        title: "Cualquier Title",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    let wrapper = shallow(<GifGrid category={category} />);

    const p = wrapper.find('p'); 
    expect(wrapper).toMatchSnapshot(); 
    expect(p.exists()).toBe(false); 
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length); 

  });
});
