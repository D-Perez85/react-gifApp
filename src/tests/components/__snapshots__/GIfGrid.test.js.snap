// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <GifGrid/> Debe de mostrar items cuando se cargan imagenes UseFetch 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Title"
      url="Cualquier URL"
    />
    <GifGridItem
      id="ABd"
      key="ABd"
      title="Cualquier Title"
      url="Cualquier URL"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en  <GifGrid/> Snapshot de GifGrid  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
